test/common/common.go

```diff
@@ -0,0 +1,446 @@
+}
+
+func LookupARecord(host, server string) ([]string, error) {
+	dnsClient := &dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	response, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Answer) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	var ips []string
+	for _, ans := range response.Answer {
+		if aRec, ok := ans.(*dns.A); ok {
+			ips = append(ips, aRec.A.String())
+		}
+	}
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	return ips, nil
+}
+
+// DnsQueryClusterInternal queries for a DNS hostname inside the VPC of a cluster using a dig pod. It returns a
+// map structure of the resolved IPs for easy lookup.
+func DnsQueryClusterInternal(ctx context.Context, t *testing.T, cl client.Client, cs *kubernetes.Clientset, namespace, hostname string) (map[string]struct{}, error) {
+	t.Helper()
+
+	clientPod := buildDigPod("digpod", namespace, hostname)
+	if err := cl.Create(ctx, clientPod); err != nil {
+		return nil, fmt.Errorf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		_ = cl.Delete(ctx, clientPod)
+	}()
+
+	// Loop until dig pod starts, then parse logs for query results.
+	var responseCode string
+	var gotIPs map[string]struct{}
+	err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(ctx context.Context) (bool, error) {
+		if err := cl.Get(ctx, types.NamespacedName{Name: clientPod.Name, Namespace: clientPod.Namespace}, clientPod); err != nil {
+			t.Errorf("Failed to get pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+			return false, nil
+		}
+		switch clientPod.Status.Phase {
+		case corev1.PodRunning:
+			t.Log("Waiting for dig pod to finish")
+			return false, nil
+		case corev1.PodPending:
+			t.Log("Waiting for dig pod to start")
+			return false, nil
+		case corev1.PodFailed, corev1.PodSucceeded:
+			// Failed or Succeeded, let's continue on to check the logs.
+			break
+		default:
+			return true, fmt.Errorf("unhandled pod status type")
+		}
+
+		// Get logs of the dig pod.
+		readCloser, err := cs.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+			Container: clientPod.Spec.Containers[0].Name,
+			Follow:    false,
+		}).Stream(ctx)
+		if err != nil {
+			t.Logf("Failed to read output from pod %s: %v (retrying)", clientPod.Name, err)
+			return false, nil
+		}
+		scanner := bufio.NewScanner(readCloser)
+		defer func() {
+			if err := readCloser.Close(); err != nil {
+				t.Fatalf("Failed to close reader for pod %s: %v", clientPod.Name, err)
+			}
+		}()
+
+		gotIPs = make(map[string]struct{})
+		for scanner.Scan() {
+			line := scanner.Text()
+
+			// Skip blank lines.
+			if strings.TrimSpace(line) == "" {
+				continue
+			}
+			// Parse status out (helpful for future debugging)
+			if strings.HasPrefix(line, ";;") && strings.Contains(line, "status:") {
```

Comment:
Why reason to not use `+short` query option which prints only the target address without anything else?